agent/dns_test.go

```diff
@@ -4688,3 +4763,24 @@ func TestDNS_Compression_Recurse(t *testing.T) {
 		t.Fatalf("doesn't look compressed: %d vs. %d", compressed, unc)
 	}
 }
+
+func TestDNSInvalidRegex(t *testing.T) {
+	tests := []struct {
+		desc    string
+		in      string
+		invalid bool
+	}{
+		{"Valid Hostname", "testnode", false},
```

Comment:
try to include edge cases like `""`, `"."`, `"a.b"`, `" "`, ...

I usually try to use generic values like `a`, `ab`, `a-b` and so forth to encode a pattern of what I'm testing. This way you don't have to invent names and you can use the pattern to group certain cases. 

Feel free to disregard both suggestions if you feel that this is sufficient.